tests/generation/test_utils.py

```diff
@@ -677,7 +679,8 @@ def test_beam_search_generate_dict_outputs_use_cache(self):
     def test_model_parallel_beam_search(self):
         for model_class in self.all_generative_model_classes:
             if "xpu" in torch_device:
-                return unittest.skip(reason="device_map='auto' does not work with XPU devices")
+                if not (is_ipex_available("2.5") or version.parse(torch.__version__) >= version.parse("2.6")):
+                    self.skipTest(reason="device_map='auto' does not work with XPU devices")
```

Comment:
BTW, I think this check could be (and better) outside the for loop.

Reply:
Indeed, check does not depend on the loop. Done in the last update.